webpack/assets/javascripts/react_app/components/common/forms/OrderableSelect/OrderableSelect.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { set } from 'lodash';
+import { TypeAheadSelect } from 'patternfly-react';
+
+import { noop } from '../../../../common/helpers';
+import { orderDragged } from './helpers';
+import OrderableToken from './components/OrderableToken';
+
+/**
+ * Wraps TypeAheadSelect with an Orderable HOC.
+ * Presumes to be wrapped in a DndProvider context.
+ */
+const OrderableSelect = ({
+  onChange,
+  defaultValue,
+  value,
+  options,
+  ...props
+}) => {
+  const [internalValue, setInternalValue] = useState(
+    (value || defaultValue)
+      .map(v => options.find(opt => opt.value === v))
+      .filter(v => !!v)
```

Comment:
this is the state's initial value, can you move it outside this scope?

```js
const initalValue =  (value || defaultValue)
      .map(v => options.find(opt => opt.value === v))
      .filter(v => !!v)

const [internalValue, setInternalValue] = useState(initalValue);
```

Reply:
I have created a custom hook for it and extracted it to a separate file entirely.